sympy/core/function.py

```diff
@@ -592,6 +599,8 @@ def bad(m):
 
         with mpmath.workprec(prec):
             v = func(*args)
+            if v == 0:
+                v = mpmath.mpf(v)
```

Comment:
Why is this needed?

Reply:
When `x1==x2`, mpmath's betainc returns an integer 0 instead of an mpmath object whose value is 0. This causes problems later on in evalf

Reply:
That sounds like a bug to be reported/fixed upstream in mpmath.

I think we should make a special `betainc_mpmath_fix`  function that wraps mpmath's routine and applies this fix. That way we can acknowledge the bug and work around it without hiding other possible bugs. The `_eval_mpmath` routine can be used to select that instead of mpmath's routine.

Reply:
I've added a function `betainc_mpmath_fix` in `beta_functions.py` and opened an issue in mpmath.